src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterFilter.cs

```diff
@@ -31,9 +33,11 @@ public void AddFilter(string providerName)
             AddFilter(providerName, Array.Empty<string>());
         }
 
-        public void AddFilter(string providerName, string[] counters)
+        public void AddFilter(string providerName, string[] counters, float? intervalSeconds = null)
```

Comment:
Renaming the IntervalSeconds property to DefaultIntervalSeconds would be more accurate now right? It appears that value is only used for counters where this interval was not provided.